src/sentry/testutils/helpers/backups.py

```diff
@@ -126,34 +127,57 @@ def reversed_dependencies():
     return sorted
 
 
+def is_control_model(model):
+    meta = model._meta
+    return not hasattr(meta, "silo_limit") or SiloMode.CONTROL in meta.silo_limit.modes
+
+
+def clear_model(model, *, use_sql: bool, reset_pks: bool):
+    using = router.db_for_write(model)
+    with unguarded_write(using=using):
+        connection = connections[using]
+
+        # For some reason, the tables for `SentryApp*` models don't get deleted properly here
+        # when using `model.objects.all().delete()`, so we have to call out to Postgres
```

Comment:
This is because of the `ParanoidManager` these models use. The `ParanoidManager` overwrites the default queryset generation to include additional conditions. The models that use `ParanoidManager` also have a `with_deleted` manager instance that will query all records.

Reply:
That makes sense! Fixed to select the correct manager instead.